Type inference error

@bobmcwhirter Could this be related to your recent changes?

[ERROR] reason: inferred type does not conform to upper bound(s)
[ERROR] inferred: org.wildfly.swarm.config.infinispan.cache_container.CustomStore<capture#99 of ?>
[ERROR] upper bound(s): org.wildfly.swarm.config.infinispan.cache_container.CustomStore<org.wildfly.swarm.config.infinispan.cache_container.CustomStore<capture#99 of ?>>,org.wildfly.swarm.config.infinispan.cache_container.CustomStore<capture#99 of ?>
[ERROR] /Users/hbraun/dev/prj/wildfly-config-api/api/target/generated-sources/org/wildfly/swarm/config/infinispan/cache_container/LocalCache.java:[614,55] incompatible types: cannot infer type arguments for org.wildfly.swarm.config.infinispan.cache_container.CustomStore<>

I saw that a lot but thought I got past it before I pushed. I'll look into
it.

--
Sent from my iPhone


On Nov 6, 2015, at 2:32 AM, Heiko Braun <notifications@github.com> wrote:

@bobmcwhirter <https://github.com/bobmcwhirter> Could this be related to
your recent changes?

[ERROR] reason: inferred type does not conform to upper bound(s)
[ERROR] inferred:
org.wildfly.swarm.config.infinispan.cache_container.CustomStore
[ERROR] upper bound(s):
org.wildfly.swarm.config.infinispan.cache_container.CustomStore>,org.wildfly.swarm.config.infinispan.cache_container.CustomStore
[ERROR]
/Users/hbraun/dev/prj/wildfly-config-api/api/target/generated-sources/org/wildfly/swarm/config/infinispan/cache_container/LocalCache.java:[614,55]
incompatible types: cannot infer type arguments for
org.wildfly.swarm.config.infinispan.cache_container.CustomStore<>

—
Reply to this email directly or view it on GitHub
<https://github.com/wildfly-swarm/wildfly-config-api/issues/11>.

--- a/generator/src/main/java/org/wildfly/swarm/config/generator/generator/ResourceFactory.java
+++ b/generator/src/main/java/org/wildfly/swarm/config/generator/generator/ResourceFactory.java
@@ -386,7 +386,7 @@ public class ResourceFactory implements SourceFactory {
             configurator.setPublic()
                     .setName(singularName)
                     .setReturnType("T")
-                    .setBody(childClassName + "<?> child = new " + childClassName + "<>(childKey);\n if ( consumer != null ) { consumer.accept(child); }\n" + singularName + "(child);\nreturn (T) this;")
+                    .setBody(childClassName + "<? extends "+childClassName+"> child = new " + childClassName + "<>(childKey);\n if ( consumer != null ) { consumer.accept(child); }\n" + singularName + "(child);\nreturn (T) this;")
                     .addAnnotation("SuppressWarnings").setStringValue("unchecked");
 
             // Add a mutator method that factories a single resource and applies a supplied configurator. Mutators are added to the containing class
@@ -490,7 +490,7 @@ public class ResourceFactory implements SourceFactory {
                     .setName(propName)
                     .setReturnType("T")
                     .setBody(
-                            childClass.getClassName() + "<?> child = new " + childClass.getClassName() + "<>();\n"
+                            childClass.getClassName() + "<? extends "+childClass.getClassName()+"> child = new " + childClass.getClassName() + "<>();\n"
                                     + "if ( consumer != null ) { consumer.accept(child); }\n"
                                     + "this.subresources." + propName + " = child;\n"
                                     + "return (T) this;"
@@ -506,7 +506,7 @@ public class ResourceFactory implements SourceFactory {
                     .setName(propName)
                     .setReturnType("T")
                     .setBody(
-                            childClass.getClassName() + "<?> child = new " + childClass.getClassName() + "<>();\n"
+                            childClass.getClassName() + "<? extends "+childClass.getClassName()+"> child = new " + childClass.getClassName() + "<>();\n"
                                     + "this.subresources." + propName + " = child;\n"
                                     + "return (T) this;"
                     )

